treeherder/log_parser/management/commands/crossreference_error_lines.py

```diff
@@ -76,8 +78,12 @@ def crossreference_error_lines(self, repository, job_guid, failure_lines,
         :param text_log_summary: text_log_summary artifact for this job
         :param bug_suggestions: Bug suggestions artifact for this job"""
 
-        summary, _ = TextLogSummary.objects.get_or_create(job_guid=job_guid,
-                                                          repository=repository)
+        if TextLogSummary.objects.filter(job_guid=job_guid).exists():
```

Comment:
I don't think this is necessary as we should now throw an exception if a text log summary already exists. Also, I think it is possible that this condition could pass, but we might still run into an integrity error if another instance of this finished before we completed this transaction. So at the very least we need to add a try...except above. I'd also personally remove this check.